arangod/Replication2/StateMachines/Prototype/PrototypeStateMachine.h

```diff
@@ -177,6 +191,8 @@ auto PrototypeLeaderState::set(Iterator begin, Iterator end)
                 core->store = core->store.set(it->first, it->second);
               }
               return idx;
+              core->lastAppliedIndex = idx;
+              core->resolvePromises(idx);
```

Comment:
Why is it allowed to resolve all promises *upto* index? Why is `idx-1` already applied?